qcodes/dataset/measurements.py

```diff
@@ -90,31 +103,56 @@ def add_result(self,
         are not unraveled but stored directly for improved performance.
 
         Args:
-            res: a dictionary with keys that are parameter names and items
-                that are the corresponding values at this measurement point.
+            res_tuple: a tuple with the first element being the parameter name
```

Comment:
Maybe we should adjust the doctring to say that one can also pass the parameter itself.